homeassistant/helpers/entity_component.py

```diff
@@ -37,6 +38,7 @@ def __init__(self, logger, domain, hass,
         self.is_polling = False
 
         self.config = None
+        self.entities_lock = threading.RLock()
```

Comment:
This should be a `threading.Lock()`. Otherwise it can still happen that the same thread claims the lock twice.